.github/workflows/build.yml

```diff
@@ -32,6 +32,25 @@ jobs:
         uses: actions/checkout@v4
         with:
           submodules: recursive
+      # Check whether gradle.properties version has been released already
+      # This prevents accidently publish same version
+      # Only check binary for arithmetic as all published together and arithmetic is the first
+      - name: Check existing release
+        run: |
+          # Check whether version contains SNAPSHOT
+          grep "^version" gradle.properties | grep -q SNAPSHOT || {
```

Comment:
does this mean that republish is allowed for *-SNAPSHOT versions?

Reply:
Yes. We need to allow publishing *-SNAPSHOT versions.

Reply:
Logic here is that we grep for SNAPSHOT and if not find move into the block for execution.